src/ui.rs

```diff
@@ -207,6 +207,30 @@ impl Ui {
         }
     }
 
+    pub fn prompt(&mut self, prompt: &str) -> io::Result<String> {
+        if !atty::is(Stream::Stdout) {
+            return Err(io::Error::new(
+                io::ErrorKind::Unsupported,
+                "not connected to a terminal",
```

Comment:
This looks like it might be unclear for the user. Something like "Cannot prompt for input since the output is not connected to a terminal"?

Reply:
Done, though this doesn't actually get displayed to the user at present (we end up falling back on the old, somewhat obscure, git error). Plausible that future refactoring might change that.

Reply:
Ah, because this is called from a callback, I see.